CMakeLists.txt

```diff
@@ -4,6 +4,17 @@
 
 cmake_minimum_required(VERSION 3.12)
 
+if((DEFINED TI_EMSCRIPTENED) AND TI_EMSCRIPTENED)
+  if(DEFINED ENV{EMSCRIPTEN_DIR})
```

Comment:
Maybe consider using CMake variables directly? (i.e. `-DEMSCRIPTEN_DIR=...`) instead of environment variables? Environment variables are quite easy to pollute.

Reply:
I like that idea, but on the other hand we use environment variables everywhere else (`TAICHI_CMAKE_ARGS`, `LLVM_DIR`,... ) so I'm not sure. @ailzhang wdyt?

Reply:
@AmesingFlank sorry for the late reply! IMHO cmake variables are preferred way and we'll try to unify all cmake args into `TAICHI_CMAKE_ARGS` ;)